staging/src/k8s.io/component-base/featuregate/feature_gate.go

```diff
@@ -296,6 +307,29 @@ func (f *featureGate) Add(features map[Feature]FeatureSpec) error {
 	return nil
 }
 
+func (f *featureGate) OverrideDefault(name Feature, override bool) error {
+	f.lock.Lock()
+	defer f.lock.Unlock()
+
+	known := map[Feature]FeatureSpec{}
+	for name, spec := range f.known.Load().(map[Feature]FeatureSpec) {
+		known[name] = spec
+	}
+
+	if spec, ok := known[name]; !ok {
+		return fmt.Errorf("cannot override default: feature %q is not registered", name)
+	} else if spec.LockToDefault && override != spec.Default {
+		return fmt.Errorf("cannot override default: feature %q default is locked to %t", name, spec.Default)
+	} else {
+		spec.Default = override
+		known[name] = spec
+	}
+
```

Comment:
should we consider when `! spec.LockToDefault` and `override == spec.Default` ?

Reply:
`SetFromMap` warns on setting Deprecated or GA features. That seems nearer to the actionable problem ("this gate will be removed, so stop making this call"), even if they're the same in practice, so I will add similar warnings here. What do you think?

Reply:
sgtm